dbcon/mysql/ha_mcs_execplan.cpp

```diff
@@ -8357,7 +8368,12 @@ int getSelectPlan(gp_walk_info& gwi, SELECT_LEX& select_lex, SCSEP& csep, bool i
         else
           rc->asc(false);
 
-        gwi.orderByCols.push_back(SRCP(rc));
+        SRCP srcprc(rc);
+        gwi.orderByCols.push_back(srcprc);
+        if (orderbyOutOfGroupBy[indexOC])
```

Comment:
Is there a guarantee `indexOC` value is less than the size of the `orderbyOutOfGroupBy`?

Reply:
Yes.

Reply:
Where exactly?